### PR TITLE
Migrate steps/location to TS

### LINKED ISSUE #37188: Migrate `./tasks/fills/steps/location.js` to TS in React admin
### Describe the solution you'd like

Convert `./tasks/fills/steps/location.js` to TS

https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/tasks/fills/steps/location.js#L34

## Acceptance Criteria

- [ ] *.js files are converted to *.ts or *.tsx
- [ ] Run `pnpm --filter=woocommerce/client/admin ts:check` without any errors
- [ ] Not break existing functionalities

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR migrates `steps/location` task to TypeScript

Closes #37188.

### How to test the changes in this Pull Request:

1. Checkout this branch and build the assets.
2. Make sure there's no TS while building the assets.
3. Smoke test OBW and other tasks.

### Other information:

-   [X] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/tasks/fills/steps/location.tsx [hunk 1]
```diff
@@ -95,9 +122,14 @@ const StoreLocation = ( {
...
 		>
-			{ ( { getInputProps, handleSubmit, setValue } ) => (
+			{ ( {
+				getInputProps,
+				handleSubmit,
+				setValue,
+			}: FormContext< FormValues > ) => (
 				<Fragment>
 					<StoreAddress
+						// @ts-expect-error
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/steps/location.tsx]: It seems that we need to add a description here to explain why the `@ts-expect-error` is necessary.

Screenshot 2023-03-17 at 13 09 10

**Review by chihsuan — DISMISSED** [review]
Thanks for working on this! @moon0326 Just left a comment to fix the lining. Other than that, looks good to me. 👍

File: plugins/woocommerce-admin/client/tasks/fills/steps/location.tsx [hunk 2]
```diff
@@ -62,26 +98,17 @@ const StoreLocation = ( {
...
+			countryState: settings?.woocommerce_default_country || '',
+			postCode: settings?.woocommerce_store_postcode || '',
 		};
 	};
 
-	const validate = ( values ) => {
-		const locale = getLocale( values.countryState );
-		const validator = getStoreAddressValidator( locale );
+	const validate = ( values: FormValues ) => {
+		const validator = getStoreAddressValidator();
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/steps/location.tsx:111]: Glad we remove this unused param!

File: plugins/woocommerce-admin/client/tasks/fills/steps/location.tsx (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/tasks/fills/steps/location.tsx]: Updated in https://github.com/woocommerce/woocommerce/pull/37257/commits/2fc4213769ab1ea316cf2f53efa8d909801ce224

**Review by rjchow — APPROVED** [review]
lgtm, thanks!

**Pull request merged**